types/header.go

```diff
@@ -77,6 +176,15 @@ func (n Nonce) MarshalText() ([]byte, error) {
 	return []byte(n.String()), nil
 }
 
+func (n *Nonce) UnmarshalText(input []byte) error {
```

Comment:
Why does this method assume that `Nonce` will be hex encoded?

Reply:
Agree i can not be sure, but to be honest not sure how would i write this different, please add suggestion :)